From bb8b53f8b3d0c365b54cb7329fd50bc57a5060ff Mon Sep 17 00:00:00 2001
From: Jonathan Helmus <jjhelmus@gmail.com>
Date: Thu, 2 Apr 2020 10:34:39 -0500
Subject: [PATCH 1/2] load log prior to calling in post.py

Load the logger prior to calling the warn method.  Adjust indentation to
match the module's style.
---
 conda_build/post.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/conda_build/post.py b/conda_build/post.py
index 782732764..1a116f48c 100644
--- a/conda_build/post.py
+++ b/conda_build/post.py
@@ -397,8 +397,9 @@ def osx_ch_link(path, link_dict, host_prefix, build_prefix, files):
 
 def mk_relative_osx(path, host_prefix, build_prefix, files, rpaths=('lib',)):
     if sys.platform != 'darwin':
-      log.warn("Found Mach-O file but patching is only supported on macOS, skipping: %s", path)
-      return
+        log = utils.get_logger(__name__)
+        log.warn("Found Mach-O file but patching is only supported on macOS, skipping: %s", path)
+        return
     prefix = build_prefix if exists(build_prefix) else host_prefix
     names = macho.otool(path, prefix)
     s = macho.install_name_change(path, prefix,

From 6632906cde0ca1693b79d4ca96eba6df6f71916f Mon Sep 17 00:00:00 2001
From: Jonathan Helmus <jjhelmus@gmail.com>
Date: Thu, 2 Apr 2020 10:48:23 -0500
Subject: [PATCH 2/2] add news item

---
 news/fix_ignore_macho.rst | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)
 create mode 100644 news/fix_ignore_macho.rst

diff --git a/news/fix_ignore_macho.rst b/news/fix_ignore_macho.rst
new file mode 100644
index 000000000..556dc381b
--- /dev/null
+++ b/news/fix_ignore_macho.rst
@@ -0,0 +1,25 @@
+Enhancements:
+-------------
+
+* <news item>
+
+Bug fixes:
+----------
+
+* load log prior to calling warn method (#3925)
+
+Deprecations:
+-------------
+
+* <news item>
+
+Docs:
+-----
+
+* <news item>
+
+Other:
+------
+
+* <news item>
+
